### PR TITLE
Move methods out of DomainLinkView

### PR DESCRIPTION
**Pull request by gherceg**
## Summary

The first 6 commits on this PR address the following: 

Preparing for new linked projects UI. The goal is to replace/modify the existing `DomainLinkView`, and this is a small step to make that easier. I _think_ this would be an ideal step regardless of a pending refactor because it makes the `DomainLinkView` easier to read. 

Moved to a `view_helpers` file and refactored in the process. Pulling out each responsibility into its own methods made this more testable, so I also added tests. This is not complete, but wanted to PR before doing more work as this PR is already too large.

In addition to those 6 commits, 2 sub-PRs were made to better isolate changes and make it easier for reviewers. You can follow those below. 

Even though none of these _should_ change the current behavior of linked projects, I requested QA to ensure this is the case.

## Feature Flag

`LINKED_DOMAINS`
## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage

Only moved code.
### QA Plan

[QA Ticket](https://dimagi-dev.atlassian.net/browse/QA-2984)
### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Review by orangejenny — APPROVED** [review]

**Comment by orangejenny** [comment]
Conveniently, this refactor also makes these methods more testable.

Some of our django apps have `helpers.py` or `view_helpers.py` files for this kind of code, but others just have helper methods in `views.py` so this is also fine as-is.

**Comment by gherceg** [comment]
Not a complete refactor. Planning on dividing up `build_other_view_models` into more specific methods for feature flag data models and bulk(?) data models. 

Also have yet to attack the other `get_master_model_status` method but that is next.

File: corehq/apps/linked_domain/tests/test_view_helpers.py [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+from corehq.apps.sms.models import Keyword
+from corehq.apps.userreports.dbaccessors import delete_all_report_configs
+from corehq.apps.userreports.models import (
+    DataSourceConfiguration,
+    ReportConfiguration,
+    ReportMeta,
+)
+
+
+def _create_report(domain, title="report", upstream_id=None, should_save=True):
```
Comment (orangejenny) [inline corehq/apps/linked_domain/tests/test_view_helpers.py:28]: I'd have made these class-level methods, part of either a mixin or superclass that would be used by the other test classes, but I think that's just a stylistic difference, I can't come up with an especially compelling argument one way or the other.

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/linked_domain/tests/test_view_helpers.py (hunk 1 above)
Comment (gherceg) [reply corehq/apps/linked_domain/tests/test_view_helpers.py:28]: Ah interesting. I had mixed feelings putting those methods where they are. I was almost positive I was writing code that already existed, but am not sure if there is a realistic solution to that problem. I think my ideal is to house methods like these in a `test_util.py` at least for each app that the tests within that app reference. If adding to a mixin or base class is more conventional within HQ, I'm happy to do that.

Comment (orangejenny) [reply corehq/apps/linked_domain/tests/test_view_helpers.py:28]: I can see a `test_util` working, HQ has a bunch of `test_util(s)` files. We do also have at least one or two text mixins - `TestXmlMixin` comes to mind. We also have some test utils that are context managers for things like creating a case for a single test. We don't have strong shared design principles across tests, so it varies a lot by django app.

No need to take action on this, I'm you'll be continuing to think about test design.

**Review by biyeun — APPROVED** [review]
this looks good on the surface to me, but deferring to jenny since she has better context of this area

**Review by orangejenny — APPROVED** [review]

**Pull request merged**